### PR TITLE
[UR] Update UR SHA to change unions to std::variant

### PR DESCRIPTION
**Pull request by hdelan**
Update UR SHA to https://github.com/oneapi-src/unified-runtime/pull/961 change unions to std::variant in the HIP and CUDA adapters.

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -3,14 +3,9 @@
...
-  #commit b38855ed815ffd076bfde5e5e06170ca4f723dc1
-  #Merge: e6343f4 6a2c548
-  #Author: Piotr Balcer <piotr.balcer@intel.com>
-  #Date:   Thu Oct 5 12:15:42 2023 +0200
-  #    Merge pull request #920 from jsji/localcopy
-  #        [UR][L0] Copy prebuilt L0 to avoid leaking shared folder path
-  set(UNIFIED_RUNTIME_TAG b38855ed815ffd076bfde5e5e06170ca4f723dc1)
+  # Do not merge!!
+  set(UNIFIED_RUNTIME_REPO "https://github.com/hdelan/unified-runtime.git")
+  set(UNIFIED_RUNTIME_TAG 2b2e2346ebe231f6155d45a67c428b2a84671a86)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:8]: Now that https://github.com/oneapi-src/unified-runtime/pull/961 is merged please pull in the latest changes from the sycl branch and update this to:

```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit 036b9cfcd8fb4be9394449b5406e6402580a63c9
  # Merge: 2ab0734 86f96f0
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Fri Oct 27 16:36:23 2023 +0100
  #     Merge pull request #961 from hdelan/change-unions-to-stdvariant
  #     [HIP][CUDA] Change unions in ur_mem_handle_t_ to stdvariant
  set(UNIFIED_RUNTIME_TAG 036b9cfcd8fb4be9394449b5406e6402580a63c9)
```

The PR title/description will need updated to reflect the contained UR changes. Then mark as ready for review.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 2]
```diff
@@ -54,16 +54,17 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
+  # Date:   Fri Oct 27 16:36:23 2023 +0100
+  #     Merge pull request #961 from hdelan/change-unions-to-stdvariant
+  #     [HIP][CUDA] Change unions in ur_mem_handle_t_ to stdvariant
+  set(UNIFIED_RUNTIME_TAG 036b9cfcd8fb4be9394449b5406e6402580a63c9)
+
+  if ("level_zero" IN_LIST SYCL_ENABLE_PLUGINS)
+    set(UR_BUILD_ADAPTER_L0 ON)
 
-  if(SYCL_PI_UR_OVERRIDE_FETCH_CONTENT_REPO)
-    set(UNIFIED_RUNTIME_REPO "${SYCL_PI_UR_OVERRIDE_FETCH_CONTENT_REPO}")
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:66]: I don't think this should be deleted

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 3]
```diff
@@ -54,16 +54,17 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
-  #         Merge pull request #965 from Bensuo/ewan/L0_internal_event_fix
-  #         [Command-buffer][L0] Reset sync-point events
-  set(UNIFIED_RUNTIME_TAG 7c8617b3710736db642585fe310a64e19e059d2e)
+  # Date:   Fri Oct 27 16:36:23 2023 +0100
+  #     Merge pull request #961 from hdelan/change-unions-to-stdvariant
+  #     [HIP][CUDA] Change unions in ur_mem_handle_t_ to stdvariant
+  set(UNIFIED_RUNTIME_TAG 036b9cfcd8fb4be9394449b5406e6402580a63c9)
+
+  if ("level_zero" IN_LIST SYCL_ENABLE_PLUGINS)
+    set(UR_BUILD_ADAPTER_L0 ON)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:66]: This doesn't feel like if should be changing here.

File: sycl/plugins/unified_runtime/CMakeLists.txt (hunk 2 above)
Comment (hdelan) [reply sycl/plugins/unified_runtime/CMakeLists.txt:66]: Will reinstate

Comment (hdelan) [reply sycl/plugins/unified_runtime/CMakeLists.txt:66]: Should be OK now

**Review by kbenzie — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/dpcpp-l0-pi-reviewers please review

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**